var/spack/repos/builtin/packages/precice/package.py

```diff
@@ -33,38 +33,68 @@ class Precice(CMakePackage):
     variant('python', default=False, description='Enable Python support')
     variant('shared', default=True, description='Build shared libraries')
 
-    # Not yet
-#    variant(
-#        'float', default=False,
-#        description='Use single precision for field data exchange')
-#    variant(
-#        'int64',
-#        default=False, description='Use 64-bit integers for indices')
-
     depends_on('cmake@3.5:', type='build')
-    depends_on('cmake@3.9.6:', type='build', when='@1.4:')
+    depends_on('cmake@3.10.2:', type='build', when='@1.4:')
     depends_on('boost@1.60.0:')
     depends_on('boost@1.65.1:', when='@1.4:')
     depends_on('eigen@3.2:')
+    depends_on('eigen@:3.3.7', type='build', when='@:1.5')  # bug in prettyprint
     depends_on('libxml2')
     depends_on('mpi', when='+mpi')
     depends_on('petsc@3.6:', when='+petsc')
-    depends_on('python@2.7', when='+python', type=('build', 'run'))
+    depends_on('python@2.7.16', when='+python', type=('build', 'run'))
+    depends_on('py-numpy@1.16.4', when='+python', type=('build', 'run'))
 
     def cmake_args(self):
         """Populate cmake arguments for precice."""
         spec = self.spec
 
+        # The xSDK installation policies were implemented after 1.5.2
+        xsdk_mode = spec.satisfies("@1.6:")
+
         def variant_bool(feature, on='ON', off='OFF'):
             """Ternary for spec variant to ON/OFF string"""
             if feature in spec:
                 return on
             return off
 
+        libxml2_includes = spec['libxml2'].headers.directories[0]
         cmake_args = [
-            '-DMPI:BOOL=%s' % variant_bool('+mpi'),
-            '-DPETSC:BOOL=%s' % variant_bool('+petsc'),
-            '-DPYTHON:BOOL=%s' % variant_bool('+python'),
             '-DBUILD_SHARED_LIBS:BOOL=%s' % variant_bool('+shared'),
+            '-DMPI:BOOL=%s' % variant_bool('+mpi'),
+
+            '-DTPL_ENABLE_BOOST=ON' if xsdk_mode else None,
```

Comment:
Does this add a `None` to the list? Might be better to conditionally append to the list.

Reply:
I was considering that too, however, it makes this section less readable. I'll look into it again.

Reply:
I agree that it is a bit strange. I pushed a clean-up commit.